### PR TITLE
chore(scene): try use RESET_VALUE

### PR DESCRIPTION
**Pull request by haweston**
## Overview
Trial of using new API option in aws-sdk/client-iottwinmaker for property update with RESET_VALUE.

Note: new API is not included and this should NOT be merged until API is released and package jsons are properly updated.

## Verifying Changes
Have a Dynamic Scene in twinMaker workspace
Do AWS Scene Composer storybook scene
Try changing in Scene Composer and confirm updates happen to Entiites in twinMaker workspace, especially clearing values to not have values any more.
Component Type to Test:

SubModelRef
Light
ModelShader
MotionIndicator
Camera
ModelRef
Overlay
Tag

### Scene Composer
For `scene-composer` package changes specifically, you can preview the component in the published storybook artifact. To do this, wait for the `Publish Storybook` action to complete below.

- Click on the workflow details
- Select the Summary item on the left
- Download the zip file

To run the storybook build locally, you need a local static web server:

```
npm install -g httpserver
cd <Extracted Zip Directory>
httpserver
```

Then open the website http://localhost:8080 to run the doc site.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: examples/react-app/package.json [hunk 1]
```diff
@@ -42,6 +42,10 @@
     ]
   },
   "devDependencies": {
+    "@aws-sdk/client-iottwinmaker": "file:~/workplace/AppKitV5/aws-sdk-client-iottwinmaker-3.571.0.tgz",
```
Comment (mumanity) [inline examples/react-app/package.json:45]: Why are you having to modify `react-app`?

File: packages/scene-composer/src/utils/entityModelUtils/lightComponent.ts [hunk 2]
```diff
@@ -76,8 +76,42 @@ export const createLightEntityComponent = (light: ILightComponent): ComponentReq
   return comp;
 };
 
-export const updateLightEntityComponent = (light: ILightComponent): ComponentUpdateRequest => {
+export const updateLightEntityComponent = (
+  light: ILightComponent,
+  oldComponent?: ILightComponent,
+): ComponentUpdateRequest => {
   const request = createLightEntityComponent(light);
+  if (oldComponent) {
```
Comment (mumanity) [inline packages/scene-composer/src/utils/entityModelUtils/lightComponent.ts:84]: You don't need this `if (oldComponent)` check as the following `if` statement covers this.

File: packages/scene-composer/src/utils/entityModelUtils/modelRefComponent.ts [hunk 3]
```diff
@@ -63,8 +63,35 @@ export const createModelRefEntityComponent = (model: IModelRefComponent): Compon
   return comp;
 };
 
-export const updateModelRefEntityComponent = (model: IModelRefComponent): ComponentUpdateRequest => {
+export const updateModelRefEntityComponent = (
+  model: IModelRefComponent,
+  oldComponent?: IModelRefComponent,
+): ComponentUpdateRequest => {
   const request = createModelRefEntityComponent(model);
+  if (oldComponent) {
```
Comment (mumanity) [inline packages/scene-composer/src/utils/entityModelUtils/modelRefComponent.ts:71]: Don't need this `if (oldComponent)` check as the subsequent checks cover the logic.

File: packages/scene-composer/src/utils/entityModelUtils/modelShaderComponent.ts [hunk 4]
```diff
@@ -34,8 +34,22 @@ export const createModelShaderEntityComponent = (shader: IColorOverlayComponent)
   return comp;
 };
 
-export const updateModelShaderEntityComponent = (modelShader: IColorOverlayComponent): ComponentUpdateRequest => {
+export const updateModelShaderEntityComponent = (
+  modelShader: IColorOverlayComponent,
+  oldComponent?: IColorOverlayComponent,
+): ComponentUpdateRequest => {
   const request = createModelShaderEntityComponent(modelShader);
+  if (oldComponent) {
```
Comment (mumanity) [inline packages/scene-composer/src/utils/entityModelUtils/modelShaderComponent.ts:42]: `if` check not needed. Covered by `(oldComponent.ruleBasedMapId || oldComponent.valueDataBinding)`.

File: packages/source-iottwinmaker/package.json [hunk 5]
```diff
@@ -63,7 +62,9 @@
     "rxjs": "^7.8.1"
   },
   "devDependencies": {
-    "@aws-sdk/types": "^3.310.0",
+    "@aws-sdk/client-iottwinmaker": "file:~/workplace/AppKitV5/aws-sdk-client-iottwinmaker-3.571.0.tgz",
```
Comment (mumanity) [inline packages/source-iottwinmaker/package.json:65]: Maybe this PR should be set to `Draft` until this is dealt with?

Comment (haweston) [reply packages/source-iottwinmaker/package.json:65]: Good idea.

File: examples/react-app/package.json (hunk 1 above)
Comment (haweston) [reply examples/react-app/package.json:45]: local library file issues.  Hopefully I can revert this in the final PR version

**Review by mumanity — APPROVED** [review]

**Pull request merged**